# HG changeset patch
# User Rik <rik@octave.org>
# Date 1390023738 28800
#      Fri Jan 17 21:42:18 2014 -0800
# Node ID 57d5bd8700df395f8273a8af613c7ee3a881e367
# Parent  515187b514113bf7472c4e6ef0e72a25705c8995
gzip.m: Re-write %!test not to depend on external system diff command.

* gzip.m: Don't rely on external 'diff' which may not exist on all operating
systems.

diff --git a/scripts/miscellaneous/gzip.m b/scripts/miscellaneous/gzip.m
--- a/scripts/miscellaneous/gzip.m
+++ b/scripts/miscellaneous/gzip.m
@@ -53,20 +53,27 @@ endfunction
 %!   [path, basename, extension] = fileparts (filename);
 %!   if (! strcmp (entry, [dirname, filesep, basename, extension, ".gz"]))
 %!     error ("gzipped file does not match expected name!");
 %!   endif
 %!   if (! exist (entry, "file"))
 %!     error ("gzipped file cannot be found!");
 %!   endif
 %!   gunzip (entry);
-%!   if (system (sprintf ("diff %s %s%c%s%s", filename, dirname, filesep,
-%!                                            basename, extension)))
+%!   fid = fopen (filename, "rb");
+%!   assert (fid >= 0);
+%!   orig_data = fread (fid);
+%!   fclose (fid);
+%!   fid = fopen ([dirname filesep basename extension], "rb");
+%!   assert (fid >= 0);
+%!   new_data = fread (fid);
+%!   fclose (fid);
+%!   if (orig_data != new_data)
 %!     error ("unzipped file not equal to original file!");
-%!   end
+%!   endif
 %! unwind_protect_cleanup
 %!   delete (filename);
 %!   delete ([dirname, filesep, basename, extension]);
 %!   rmdir (dirname);
 %! end_unwind_protect
 
 %!error gzip ()
 %!error gzip ("1", "2", "3")
